pkg/pkgmgr/rpm.go

```diff
@@ -234,28 +235,45 @@ func (rm *rpmManager) probeRPMStatus(ctx context.Context, toolImage string) erro
 		filepath.Join(rpmManifestPath, rpmManifest1),
 		filepath.Join(rpmManifestPath, rpmManifest2),
 	}
-	rpmDBStr := strings.Join(rpmDBList, " ")
-	const probeDBTemplate = `/usr/sbin/busybox sh -c '
-		DB_LIST="%s"; n=1;
-		while [ $n -le %d ];
-		do DB=$(echo "$DB_LIST" | /usr/sbin/busybox cut -d " " -f $n);
-			echo $DB;
-			if [ -f $DB ];
-			then /usr/sbin/busybox cp $DB %s;
-			fi;
-			n=$(($n+1));
-		done'`
-	probeDBCmd := fmt.Sprintf(probeDBTemplate, rpmDBStr, len(rpmDBList), resultsPath)
 
-	probed := mkFolders.Run(llb.Shlex(probeToolsCmd)).Run(llb.Shlex(probeDBCmd)).Root()
+	// probed := mkFolders.Run(llb.Shlex(probeToolsCmd)).Run(llb.Shlex(probeDBCmd)).Root()
+	toolListPath := filepath.Join(resultsPath, "tool_list")
+	dbListPath := filepath.Join(resultsPath, "rpm_db_list")
+
+	probed := buildkit.WithArrayFile(&mkFolders, toolListPath, toolList)
+	probed = buildkit.WithArrayFile(&probed, dbListPath, rpmDBList)
+	probed = probed.Run(llb.Args([]string{
+		`/usr/sbin/busybox`, `env`,
+		buildkit.Env("TOOL_LIST_PATH", toolListPath),
```

Comment:
Same comment about env

Reply:
Done

Reply:
It all depends on what you are returning.
Each state modification leaves the last state alone.
You can also add it to just the run step.

Reply:
> You can also add it to just the run step.

Ah, I missed that that was an option. That's what I was looking for!